guides/source/active_record_multiple_databases.md

```diff
@@ -433,6 +433,8 @@ ActiveRecord::Base.connected_to(role: :reading, shard: :shard_one) do
 end
 ```
 
+Note: Do not use the same database in different shard definitions. Doing so can cause `LockWaitTimeout` errors to appear sporadically.
```

Comment:
I'm not sure we should add this. There's SO many notes/warnings we could add to the docs about things you shouldn't do. For example we could say "dont try to write to readonly connections" but we don't because that's relatively understood. I'm not sure why you'd have shards at all if they are all pointing to the same database. I get you hit this in your app but I don't know if it's a common enough mistake to add to the docs.